crates/core/app.rs

```diff
@@ -1185,6 +1193,36 @@ This flag can be disabled with --no-encoding.
     args.push(arg);
 }
 
+fn flag_engine(args: &mut Vec<RGArg>) {
+    const SHORT: &str = "Specify which regexp engine to use: default|pcre2|auto-hybrid.";
+    const LONG: &str = long!(
+        "\
+Specify which regular expression engine to use. When you choose a
+regex engine, it applies that choice for every regex provided to ripgrep (e.g.,
+via multiple -e/--regexp or -f/--file flags).
+
+Accepted values are `default|pcre2|auto-hybrid`.
+
+The default value is default, which is the fastest and should be good for most
+use-cases. PCRE2 engine is generally useful when you want to use features such
+as look-around backreferences. auto-hybrid will dynamically choose between
+supported regex engines depending on the features used in a pattern.
+
+Please have a look at the custom flags '--pcre2' and '--auto-hybrid-regexp'
+which explain their respective meaning in more depth. Those flags override
+any value provided to the `engine` option.
+
+Using '--pcre2' or '--auto-hybrid-regexp' overrides the value of this flag.
+"
+    );
+    let arg = RGArg::flag("engine", "ENGINE")
+        .help(SHORT)
+        .long_help(LONG)
+        .possible_values(&["default", "pcre2", "auto-hybrid"])
+        .default_value("default");
```

Comment:
Hmm, so I think this is not quite right. I think we want this to explicitly say that it overrides the `--pcre2` and `--auto-hybrid-regex` flags. (And vice versa for them.) Otherwise, something like `--pcre2 --engine default` would still use PCRE2 instead of what it should do: use the default engine.

Reply:
OK, fair point. 

While testing now I was unsure to add the `engine` override for the `no-pcre2` and `no-auto-hybrid-regexp` since intuitively the option set `--engine=pcre2 --no-auto-hybrid-regex` should enable pcre2.

But since `--pcre2 --no-auto-hybrid-regex` doesn't enable `pcre2` neither, I guess it's just better to keep backward compatibility with default options and add overrides everywhere.

Reply:
OK, fair point. 

While testing now I was unsure to add the `engine` override for the `no-pcre2` and `no-auto-hybrid-regexp` since intuitively the option set `--engine=pcre2 --no-auto-hybrid-regex` should enable pcre2.

But since `--pcre2 --no-auto-hybrid-regex` doesn't enable `pcre2` neither, I guess it's just better to keep backward compatibility with default options and add overrides everywhere.

Reply:
Yeah good point. Defer to backward compatibility for now. I'll take another pass over this before the next release to make sure it all makes sense.